### DISCUSSION TITLE
Make GraphEdit more flexible

### DISCUSSION THREAD

**Discussion by lostdisplay** [godotengine/godot-proposals#7550] [GUI]
### Describe the project you are working on

A planning app like [milanote](https://milanote.com/) or [masterplan](https://solarlune.itch.io/masterplan)

### Describe the problem or limitation you are having in your project

GraphEdit being too inflexible.

It's impossible, requires messing with internal nodes or is really sketchy to do to:
- Hide/disable the UI in the upper-left corner
- Hide/disable the scrollbars
- Make the grid infinite
- Change the form/design of the background
- Make the dragging and zoom animatable (Like Camera2Ds position smoothing)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adding options in the inspector

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Almost everything mentioned above is possible by messing with internal nodes. Haven't looked at dragging/zoom smoothing just yet.

### Is there a reason why this should be core and not an add-on in the asset library?

I think giving the user the option to disable UI elements should be considered. Making the grid infinite, custom backgrounds and dragging/zoom smoothing are probably things that shouldn't be integrated into core but at least should not require messing with internal nodes. :)

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

A refactor of GraphEdit is being done at https://github.com/godotengine/godot/pull/79308 (merged in 4.2), https://github.com/godotengine/godot/pull/67152 and https://github.com/godotengine/godot/pull/79311 (which still need to be completed).

PS: In the future, please open one proposal per requested feature. This is especially the case when features can be implemented separately from each other (e.g. camera smoothing).
